libsolidity/interface/StandardCompiler.cpp

```diff
@@ -1065,12 +1065,15 @@ std::variant<StandardCompiler::InputsAndSettings, Json::Value> StandardCompiler:
 	return { std::move(ret) };
 }
 
-Json::Value StandardCompiler::compileSolidity(StandardCompiler::InputsAndSettings _inputsAndSettings)
+unique_ptr<CompilerStack> StandardCompiler::createCompilerStack(
+	ReadCallback::Callback _readFile,
+	StandardCompiler::InputsAndSettings const& _inputsAndSettings
+)
 {
-	CompilerStack compilerStack(m_readFile);
+	auto compilerStackPtr = make_unique<CompilerStack>(_readFile);
```

Comment:
CompilerStack has a check that prevents two instances to be alive at the same time, so I don't think this function is a good idea. I know you had this as a first version, but we need a way to apply the settings to the compiler stack instead.

The StandardCompiler was not meant to be usend in a long-lived process..

Reply:
Why do we need the StandardCompiler at all for the LSP? The only two thing in the settings that are not related to bytecode generation is remappings and sources and I'm not sure if that warrants such a refactoring...